connectors/src/connectors/slack/auto_join_channel.ts

```diff
@@ -0,0 +1,81 @@
+import type { Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+
+import { joinChannel } from "@connectors/connectors/slack/lib/channels";
+import { getSlackClient } from "@connectors/connectors/slack/lib/slack_client";
+import {
+  SlackChannel,
+  SlackConfigurationModel,
+} from "@connectors/lib/models/slack";
+import type { Logger } from "@connectors/logger/logger";
+import { ConnectorResource } from "@connectors/resources/connector_resource";
+
+export function isChannelNameWhitelisted(
+  remoteChannelName: string,
+  whiteListedChannelPattern?: string | null
+): boolean {
+  if (!whiteListedChannelPattern || whiteListedChannelPattern === "") {
+    return false;
+  }
+
+  const regex = new RegExp(whiteListedChannelPattern);
+  return regex.test(remoteChannelName);
+}
+
+export async function autoJoinChannel(
+  teamId: string,
+  logger: Logger,
+  requestedConnectorId?: string,
+  slackChannelId?: string
+): Promise<Result<undefined, Error>> {
+  const slackConfiguration = await SlackConfigurationModel.findOne({
+    where: {
+      slackTeamId: teamId,
+    },
+  });
+  if (!slackConfiguration || !slackChannelId) {
```

Comment:
Should ideally be a different error message when slackChannelId is undefined otherwise we won't be able to debug

Reply:
If not at least add a lot that prints the two